packages/uix-host-react/src/components/Extensible.tsx

```diff
@@ -122,7 +114,12 @@ export function Extensible({
     return () => {
       host.unload().catch(logError("Unload of extensions failed!"));
     };
-  }, [host, extensions]);
+  }, [debug, hostName, runtimeContainer, sharedContext, extensions]);
+
+  // skip render before host is initialized
+  if (!host) {
```

Comment:
Not sure if it makes sense, but here are my two cents:
- I would love to see an extensible UI that doesn't block main UI rendering, so it can start doing its things earlier. Initial render returns no children & we wrap almost the entire application with `<Extensible>` component. Could you please check if we can render children on initial load?
- Hopefully, react is smart enough to avoid unnecessary re-renders, just want to confirm that non-extensible and extensible components are not refreshed if there are no changes in `extensions` list. Also, if there are some changes, it will be nice to see re-renders for only added/removed extensible components.

Reply:
These are good observations and we should implement them as tests.

Reply:
At first I tried to render the children with an empty set of extensions, like @kokoc said. I see your concern, which is that you don't want to wait for the extension registry. But like @skolodyazhnyy said, it's not waiting for a fetch, it's just waiting for one useEffect.

My original intent was to render the children with a Host that exists, but is empty of extensions. Since the Host maintains its identity as extensions are loaded, it should allow React to re-render less; ideally, only the descendent components that receive a new set of extensions need to re-render.

This is hard to test without proper unit/integration tests that can introspect React's render cycle, but for now I think we can check re-render based on perceived load (does it feel like it sucks, etc.)